concourse/steps/helmcharts.mako

```diff
@@ -0,0 +1,191 @@
+
+main_repo_relpath = job_variant.main_repository().resource_name()
+effective_version_path = os.path.join(job_step.input('version_path'), 'version')
+%>
+import datetime
+import hashlib
+import json
+import logging
+import os
+import subprocess
+import tempfile
+
+import jsonpath_ng
+import yaml
+
+import ccc.oci
+import oci.model
+import concourse.steps.component_descriptor_util as cdu
+import oci.model
+
+logger = logging.getLogger('helmcharts.step')
+
+with open('${effective_version_path}') as f:
+  effective_version = f.read().strip()
+
+component_descriptor = cdu.component_descriptor_from_dir(
+  '${job_step.input('component_descriptor_dir')}',
+)
+component = component_descriptor.component
+
+logger.info("Publishing ${len(helmchart_cfgs)} helmchart(s):")
+% for helmchart_cfg in helmchart_cfgs:
+logger.info("${helmchart_cfg}")
+print()
+% endfor
+
+oci_client = ccc.oci.oci_client()
+
+% for helmchart_cfg in helmchart_cfgs:
+logger.info('Chart: {helmchart_cfg.name}')
+print()
+helmchart_dir = os.path.join('${main_repo_relpath}', '${helmchart_cfg.dir}')
+helmchart_name = '${helmchart_cfg.name}'
+print(f'{helmchart_dir=} {helmchart_name=}')
+if not os.path.isdir(helmchart_dir):
+  logger.error(f'not an existing directory: {helmchart_dir=}')
+  exit(1)
+helmchart_outdir = 'helmchart-archives.d'
+os.mkdir(helmchart_outdir)
+
+# preprocess helmchart
+# - overwrite helmchart-name according to pipeline-cfg (to ensure consistency between content
+#   and target-ref
+# - overwrite values in values.yaml (in particular to inject image-references from current build)
+values_yaml_path = os.path.join(helmchart_dir, 'values.yaml')
+with open(values_yaml_path) as f:
+  values = yaml.safe_load(f)
+
+def find_resource(name: str):
+  for resource in component.resources:
+    if resource.name == name:
+      return resource
+  logger.error(f'did not find resource with name {name} in component-descriptor')
+  exit(1)
+
+%  for mapping_cfg in helmchart_cfg.mappings:
+ref = '${mapping_cfg.ref}'.removeprefix('ocm-resource:')
+image_name, image_attr = ref.split('.')
+resource = find_resource(name=image_name)
+
+image_ref = oci.model.OciImageReference(resource.access.imageReference)
+if image_attr == 'repository':
+  value = image_ref.ref_without_tag
+elif image_attr == 'tag':
+  value = image_ref.tag
+elif image_attr == 'image':
+  value = str(image_ref)
+
+attribute = '${mapping_cfg.attribute}'
+attribute_path = jsonpath_ng.parse(attribute)
+
+logger.info(f'resolved {ref} to {value}. Patching values.yaml')
+attribute_path.update_or_create(values, value)
+%  endfor
+
+with open(values_yaml_path, 'w') as f:
+  yaml.safe_dump(values, f)
+
+charts_yaml = os.path.join(helmchart_dir, 'Chart.yaml')
+with open(charts_yaml) as f:
+  charts_yaml_values = yaml.safe_load(f)
+
+if charts_yaml_values.get('name') != helmchart_name:
+  logger.warning(f'{charts_yaml}.name differs from {helmchart_name} - patching')
+  charts_yaml_values['name'] = helmchart_name
```

Comment:
Shouldn't we also patch the chart's version?

Reply:
that attribute is patched by `helm package` via the `--version`-argument, if I am not mistaken.

Reply:
ah that makes sense, ty